### DISCUSSION TITLE
Can I implement AutoEncoder using River ?

### DISCUSSION THREAD

**Discussion by ercangunbilek** [online-ml/river#974] [Q&A] [answered]
I want to establish an auto encoder model for my streaming data project. How can I do it ?

**Comment by kulbachcedric** [comment, marked as answer]
Hi @ercangunbilek,

we are currently working on implementing AEs in river. Actually you can train AEs based on PyTorch with 
[DeepRiver](https://github.com/kulbachcedric/DeepRiver). It is fully compatible with river. However, it has still some bugs and will be moved soon to the online-ml group. 

Best

**Reply by ercangunbilek** [reply]
Hi, @kulbachcedric ,
Thank you for your suggestions. I am working master thesis now. I want to use DeepRiver but you said that "it has still some bugs".
What can I do now ? Is DeepRiver a risk for my master thesis ? Can you share me other tools and library for streaming auto encoder ?
What are the best practices for the subject ?

**Reply by MaxHalford** [reply]
Hey @ercangunbilek.

> Is DeepRiver a risk for my master thesis ?

I think "risk" is an exaggeration. What Cedric developed is essentially a wrapper on top of PyTorch to make it compatible with River. Even if it has bugs, I'm sure they're easy to fix. Most of the hard work will still be to write your PyTorch models.
